### PR TITLE
install pysqlite for devstack and jenkins workers

### PR DESCRIPTION
**Pull request by estute**
## Configuration Pull Request

Make sure that the following steps are done before merging
- [ ] @devops team member has commented with :+1:
- [ ] are you adding any new default values that need to be overriden when this goes live?  
  - [ ] Open a ticket (DEVOPS) to make sure that they have been added to secure vars.
  - [ ] Add an entry to the CHANGELOG.

### CODE REVIEW

**Comment by estute** [comment]
@jz @cpennington

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 1]
```diff
@@ -151,6 +151,10 @@
     - install
     - install:app-requirements
 
+# Override default dependencies for installing on a devstack
+- include_vars: ../vars/devstack.yml
+  when: devstack == true
```
Comment (cpennington) [inline playbooks/roles/edxapp/tasks/deploy.yml]: This is the one part of this PR that seems questionable to me. It seems like we shouldn't be adding explicit references to devstack specific things into the generic deploy role (and should instead be adding the devstack specific things to the devstack role or to code that actually builds the devstack, as an override).

**Comment by estute** [comment]
adding @jibsheet for a devops opinion

**Comment by estute** [comment]
This change is related to https://github.com/edx/edx-platform/pull/13826

**Review by cpennington — APPROVED** [review]
It's too bad that YAML doesn't have a way to append to lists, so that we have to duplicate the full set of requirements files.

**Review by jibsheet — APPROVED** [review]
Per some offline conversation with @estute - as long as we can still cut edx-platform AMIs (slightly different build process ) then this is unfortunate but fine.  I did a quick bit of digging on ansible workarounds, but they're all dictionary based, not list based.

**Comment by estute** [comment]
AMIs cut successfully.

**Pull request merged**